modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/DefaultArchiveDeployer.java

```diff
@@ -303,7 +303,8 @@ public void deployArtifact(T artifact) throws DeploymentException {
   private void addZombieApp(Artifact artifact) {
     if (allResourcesExist(artifact.getResourceFiles())) {
       try {
-        artifactZombieMap.put(artifact.getArtifactName(), new ZombieArtifact(artifact.getResourceFiles()));
+        artifactZombieMap.put(artifact.getArtifactName(),
```

Comment:
When the artifact has redeploymentEnabled=false do not include the descriptor in the list